test-files/test_gap_9163_lazy_regex_semantics.ts

```
@@ -0,0 +1,120 @@
+function mk() {
+  return /dup/g;
+}
+const p = mk();
+const q = mk();
+console.log("distinct:" + (p !== q));
+p.lastIndex = 3;
+console.log("independent-lastIndex:" + (q.lastIndex === 0));
+(p as unknown as { tag?: string }).tag = "first";
+console.log("independent-expando:" + ((q as unknown as { tag?: string }).tag === undefined));
+// Two never-matched siblings are still distinct.
+console.log("distinct-unmatched:" + (/never/ !== /never/));
```

Biome (2.5.7)

[error] This comparison uses the same expression on both sides.

(lint/suspicious/noSelfCompare)